src/main/kotlin/org/jitsi/rtp/rtcp/RtcpSdesPacket.kt

```diff
@@ -149,7 +149,15 @@ abstract class SdesItem(
             buf.getByteAsInt(baseOffset + LENGTH_OFFSET)
 
         fun copyData(buf: ByteArray, baseOffset: Int, dataLength: Int): ByteArray {
-            val copy = BufferPool.getArray(dataLength)
+            if (dataLength <= 0) {
+                return ByteArrayUtils.emptyByteArray
+            }
+            val copy = when {
+                // Don't use the pool for small arrays.
+                dataLength < 30 -> ByteArray(dataLength)
```

Comment:
I'm not sure I follow why we wouldn't use the pool for small arrays: it's not so much the size we care about as having to frequently reallocate the objects, so that seems like it'd apply here?

Reply:
It is about the size. Allocating an object of any size is really cheap: it just bumps a pointer. The problem is that once the Eden space is full it triggers a GC, so we want to reduce the allocation rate (in bytes).

The minimum size that the pool allocates is 220 bytes, so I thought it's inefficient to use it just for these items (they should be a few bytes each right?). But I don't really know what the tradeoff is. I can revert this and always use the pool.

Reply:
But if we've got lots of places allocating small arrays, we can still run into the same problem, no?

Reply:
Ok, I think I'm not well-versed enough in the tradeoffs there to have an informed opinion, but I wonder, if we do want to do this, should we still use the pool for consistency and have the pool decide to allocate the buffer rather than pulling from the pool?  Then, if we decide we want to tweak the threshold or whatever we can do it in a single place?  For this I think the pool would also ignore returned buffers over the threshold.  Don't know if this would make bookkeeping awkward.

Reply:
Yeah, I think that's a good idea